src/types/Wallet/Wallet.js

```
@@ -114,10 +121,15 @@ class Wallet extends EventEmitter {
 
     this.store = this.storage.store;
 
-    if (this.unsafeOptions.skipSynchronizationBeforeHeight) {
+    if (createdFromNewMnemonic) {
       // As it is pretty complicated to pass any of wallet options
       // to a specific plugin, using `store` as an options mediator
       // is easier.
+
```

Let's throw an exception if `this.unsafeOptions.skipSynchronizationBeforeHeight` specified?